### PR TITLE
pythia: 8.244 -> 8.303

### PR DESCRIPTION
**Pull request by veprbl**
###### Motivation for this change

Bump pythia version to the latest version

Closes: #90120

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -26755,7 +26755,9 @@ in
 
   nlojet = callPackage ../development/libraries/physics/nlojet { };
 
-  pythia = callPackage ../development/libraries/physics/pythia { };
+  pythia = callPackage ../development/libraries/physics/pythia {
+    hepmc = hepmc2;
```
Comment (drewrisinger) [inline pkgs/top-level/all-packages.nix:26759]: Why do you lock ``hepmc = hepmc2`` if ``pythia`` seems to support ``hepmc3`` (i.e. has configure flags for hepmc3)?

**Review by drewrisinger — COMMENTED** [review]
* [x] Diff LGTM minus comment on hepmc version locking.
* [x] Commits LGTM
* [x] Builds via ``nix-review`` on ``x86_64-linux``:

```
https://github.com/NixOS/nixpkgs/pull/99993
2 packages built:
pythia sacrifice
```

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (veprbl) [reply pkgs/top-level/all-packages.nix:26759]: Have not had time to test `hepmc3` yet. I am using the `sacrifice` interface which only works with `hepmc2`. There is no real rush to switch now.

Comment (drewrisinger) [reply pkgs/top-level/all-packages.nix:26759]: Maybe add a comment to that effect about why the version is restricted

Comment (veprbl) [reply pkgs/top-level/all-packages.nix:26759]: It's not restricted, just that the default is unchanged. `pythia.override { hepmc=hepmc3; }` will compile fine.

**Review by drewrisinger — APPROVED** [review]

**Pull request merged**